docs/enhancements/dynamic-fetching-of-openshift-versions.md

```diff
@@ -0,0 +1,280 @@
+
+As a user of assisted-installer UI, I should have a list of the latest versions
+for each minor release.
+
+#### Installing an Older Version of a z-stream
+
+As a user of assisted-installer UI, I should have the option to browse the
+different available versions that are fetched from upgrade channels.
+
+#### Configuring the Dynamically-loaded Versions of a Stream
+
+As a system administrator for the assisted-installer, I should have the option
+to configure both dynamically-loaded and statically-loaded OpenShift version.
+To do that, I can provide the following environment variables for example:
+```
+RELEASE_IMAGES='[
+  {
+    "openshift_version": "4.12-multi",
+    "cpu_architecture": "multi",
+    "cpu_architectures": [
+        "x86_64",
+        "arm64",
+        "ppc64le",
+        "s390x"
+    ],
+    "url": "quay.io/openshift-release-dev/ocp-release:4.12.0-rc.4-multi",
+    "version": "4.12.0-rc.4"
+  }
+]'
+
+DYNAMIC_OPENSHIFT_RELEASES_CONFIG='[
+  {
+    "openshift_version": "4.11",
+    "upgrade_channels": [
+        {
+          "cpu_architecture": "x86_64",
+          "channels": ["stable"]
+        }
+    ]
+  },
+  {
+    "openshift_version": "4.11",
+    "upgrade_channels": [
+      {
+        "cpu_architecture": "x86_64",
+        "channels": ["candidate", "stable"]
+      },
+      {
+        "cpu_architecture": "arm64",
+        "channels": ["candidate", "stable"]
+      }
+    ]
+  }
+]'
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+Mostly conveyed in ``Risks and Mitigations``.
+
+### Risks and Mitigations
+
+We have recently seen occasions where the disk space is being exhausted due to
+the assisted-service caching a lot of different openshift installer binaries.
+It is important to make sure we don't make this problem more severe, as this
+enhancement introduces much more versions which may (or may not) cause use of
+them.
+A possible solution here is implementing the cache as a LRU-like cache, making
+sure we evict the old entries when we reach a certain capacity.
+
+We should make sure we don't consume too much memory when fetching releases.
+It's especially critical when the user queries for the different versions, but
+might happen in other cases too.
+
+We need to think of the retention policy of versions. We don't expect to have
+versions removed from the upgrade-channels API, but just in case we should only
+add versions and not mirror exactly the given result of those API responses.
+As opposed to that, when a minor release gets EOL (end-of-life) and we remove it
+from the configuration, the service should remove all the stale versions from
+the database.
+
+## Design Details [optional]
+
+If an enhancement is complex enough, design details should be included. When not
+included, reviewers reserve the right to ask for this section to be filled in to
+enable more thoughtful discussion about the enhancement and it's impact.
+
+### Open Questions
```

Comment:
inst our main limitation is that the image service downloads all the images upon restart? so every release we'll have to restart the service?

Reply:
in this suggestion the image service stays static with what images it provides (that are x.y images for each supported cpu-arch)

Reply:
so we'll use outdated live iso?

Reply:
no, we have an existing process to update rhcos images from mirror.openshift.com